src/tabs/tab.directive.ts

```diff
@@ -12,13 +12,27 @@ import {
 } from '@angular/core';
 import { TabsetComponent } from './tabset.component';
 
-@Directive({ selector: 'tab, [tab]' })
+let id = 1;
+let tabpanelId = 1;
+
+@Directive({
+  selector: 'tab, [tab]',
+  host: {
+    role: 'tabpanel'
+  }
+})
 export class TabDirective implements OnInit, OnDestroy {
   /** tab header text */
   @Input() heading: string;
   /** tab id. The same id with suffix '-link' will be added to the corresponding &lt;li&gt; element  */
+  @HostBinding('attr.aria-labelledby')
+  @HostBinding('attr.id')
+  @Input() id = `tab-${id++}-link`;
+
+  /** tabpanel id  */
   @HostBinding('attr.id')
-  @Input() id: string;
+  @Input() tabpanelId = `tabpanel-${tabpanelId++}`;
```

Comment:
get rid of this and use only one `id`, with suffix `-link` (for `<a>` id and for `aria-labeledby`)